tests/unit/nutrition.t

```diff
@@ -7,1017 +7,765 @@ use Test2::V0;
+								value => 30,
+								unit => "kcal",
+							}
+						}
+					}
+				]
+			}
+		}
+	],
+	[
+		# Generated set should have converted per when different per in nutrients given nutrient with most priority with per in 100g/ml
+		"convert_per_nutrients_when_wanted_per_100g",
+		{
+			nutrition => {
+				nutrient_sets => [
+					{
+						preparation => "as_sold",
+						per => "100g",
+						per_quantity => "100",
+						per_unit => "g",
+						source => "packaging",
+						nutrients => {
+							sodium => {
+								value_string => "2.5",
+								value => 2.5,
+								unit => "g",
+							}
+						}
+					},
+					{
+						preparation => "as_sold",
+						per => "serving",
+						per_quantity => "50",
+						per_unit => "g",
+						source => "packaging",
+						nutrients => {
+							sugars => {
+								value_string => "6.3",
+								value => 6.3,
+								unit => "g",
+							}
+						}
+					}
+				]
+			}
+		}
+	],
+	[
+		# Generated set should have converted per when different per in nutrients given nutrient with most priority with per in serving
+		"convert_per_nutrients_when_wanted_per_serving",
+		{
+			nutrition => {
+				nutrient_sets => [
+					{
+						preparation => "as_sold",
+						per => "100g",
+						per_quantity => "100",
+						per_unit => "g",
+						source => "packaging",
+						nutrients => {
+							"sodium" => {
+								value_string => "2.5",
+								value => 2.5,
+								unit => "g",
+							}
+						}
+					},
+					{
+						preparation => "as_sold",
+						per => "serving",
+						per_quantity => "50",
+						per_unit => "g",
+						source => "manufacturer",
+						nutrients => {
+							"sugars" => {
+								value_string => "6.3",
+								value => 6.3,
+								unit => "g",
+							}
+						}
+					}
+				]
+			}
+		}
+	],
+	[
+		# Generated set should have converted per when different per in nutrients given nutrients per in different servings
+		"convert_per_nutrients_when_different_per_serving_quantities",
+		{
+			nutrition => {
+				nutrient_sets => [
+					{
+						preparation => "as_sold",
+						per => "serving",
+						per_quantity => "100",
+						per_unit => "g",
+						source => "packaging",
+						nutrients => {
+							"sodium" => {
+								value_string => "2.5",
+								value => 2.5,
+								unit => "g",
+							}
+						}
+					},
+					{
+						preparation => "as_sold",
+						per => "serving",
+						per_quantity => "50",
+						per_unit => "g",
+						source => "manufacturer",
+						nutrients => {
+							"sugars" => {
+								value_string => "6.3",
+								value => 6.3,
+								unit => "g",
+							}
+						}
+					}
+				]
+			}
+		}
+	],
+	[
+		# Generated set should have converted per nutrients given nutrients per in different serving units with wanted in g
+		"convert_per_nutrients_when_different_per_units_g",
+		{
+			nutrition => {
+				nutrient_sets => [
+					{
+						preparation => "as_sold",
+						per => "serving",
+						per_quantity => "100",
+						per_unit => "mg",
```

Comment:
This test is a bit weird because we have 2.5 g of sodium in a 50 mg serving. Can you change it to possible values? When I was looking at the results I thought there was a mistake.